extensions/applicationinsights-osplugin-js/src/OsPlugin.ts

```diff
@@ -0,0 +1,253 @@
+/**
+* OsPlugin.ts
+* @author Siyu Niu (siyuniu)
+* @copyright Microsoft 2024
+*/
+import dynamicProto from "@microsoft/dynamicproto-js";
+import { Extensions, IConfig } from "@microsoft/applicationinsights-common";
+import {
+    BaseTelemetryPlugin, IAppInsightsCore, IConfigDefaults, IConfiguration, IPlugin, IProcessTelemetryContext,
+    IProcessTelemetryUnloadContext, ITelemetryItem, ITelemetryUnloadState, _eInternalMessageId, _throwInternal, addPageHideEventListener,
+    addPageUnloadEventListener, arrForEach, createProcessTelemetryContext, createUniqueNamespace, eLoggingSeverity, getSetValue,
+    mergeEvtNamespace, onConfigChange, removePageHideEventListener, removePageUnloadEventListener, setValue
+} from "@microsoft/applicationinsights-core-js";
+import { doAwaitResponse } from "@nevware21/ts-async";
+import { objDeepFreeze, objDefineAccessors } from "@nevware21/ts-utils";
+import { IOSPluginConfiguration } from "./DataModels";
+
+const defaultgetOSTimeoutMs = 5000;
+const strExt = "ext";
+const maxRetry = 3;
+interface platformVersionInterface {
+    brands?: { brand: string, version: string }[],
+    mobile?: boolean,
+    platform?: string,
+    platformVersion?: string
+}
+interface UserAgentHighEntropyData {
+    platformVersion: platformVersionInterface
+}
+interface ModernNavigator {
+    userAgentData?: {
+      getHighEntropyValues?: (fields: ["platformVersion"]) => Promise<UserAgentHighEntropyData>;
+    };
+  }
+const defaultOSConfig: IConfigDefaults<IOSPluginConfiguration> = objDeepFreeze({
+    getOSTimeoutMs: defaultgetOSTimeoutMs,
+    endpointIsBreeze: true
```

Comment:
Lets not call this the "internal" endpoint name.

How about something like mergeOsNameVersion, also lets have it default to `undefined` and we can add some code to automatically "detect" whether the Post channel or Sender is installed

`core.getPlugin("<sender or post channel name>").plugin !== null`

This way users don't really need to set this.